buildSrc/src/main/java/Dependencies.kt

```diff
@@ -11,7 +11,7 @@ object BuildPluginsVersion {
 
 object LibsVersion {
     const val JUNIT = "4.13.2"
-    const val ASM = "9.2"
+    const val ASM = "7.0" // compatibility matrix -> https://developer.android.com/reference/tools/gradle-api/7.1/com/android/build/api/instrumentation/InstrumentationContext#apiversion
```

Comment:
If we're using AGP 7.1.x, it should be ASM9, no?

Reply:
If this is a problem, we can add the ASM version to the CI matrix, and read it as `VERSION_AGP`

Reply:
so, we actually get the ASM version from the AGP, when the plugin is running on the clients projects (hence, the compileOnly change for ASM). So no matter what we use here, it will be overridden by the gradle's dependency resolution to what AGP is using.

I just defined a min. supported version here (as the min support AGP version is 7.0.0 -> ASM 7.0). So it's similar to our kotlin compatibility approach.

Basically, our sample apps would be built against what ASM version AGP is using underneath, so the current CI matrix that we have for AGP versions is sufficient.

Reply:
gotcha, makes sense.